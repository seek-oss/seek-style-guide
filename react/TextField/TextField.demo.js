import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TextField from './TextField';
import styles from './TextField.less';
import * as sketch from './TextField.sketch';
import classnames from 'classnames';
import fieldMessageOptions from '../FieldMessage/FieldMessage.demoFragment';
import fieldLabelOptions from '../FieldLabel/FieldLabel.demoFragment';

class TextFieldContainer extends Component {
  static propTypes = {
    component: PropTypes.func.isRequired,
    componentProps: PropTypes.object.isRequired
  };

  constructor() {
    super();

    this.state = {
      inputValue: ''
    };
  }

  handleChange = event => {
    this.setState({
      inputValue: event.target.value
    });
  };

  handleClear = () => {
    this.setState({
      inputValue: ''
    });
  };

  render() {
    const { component: DemoComponent, componentProps } = this.props;
    const { inputValue } = this.state;

    return (
      <div style={{ width: '300px' }}>
        <DemoComponent
          {...componentProps}
          onChange={this.handleChange}
          value={inputValue}
          onClear={this.handleClear}
        />
      </div>
    );
  }
}

export default {
  route: '/textfield',
  title: 'Text Field',
  category: 'Form',
  component: TextField,
  container: TextFieldContainer,
  sketch,
  initialProps: {
    id: 'firstName',
    label: 'First Name',
    message: 'e.g. Olivia',
    // Documentation only:
    onChange: () => {},
    value: '...',
    onClear: () => {}
  },
  options: [
    {
      label: 'States',
      type: 'checklist',
      states: [
        {
          label: 'Focus',
          transformProps: ({ className, ...props }) => ({
            ...props,
            className: classnames(className, styles.rootFocus)
          })
        }
      ]
    },
    ...fieldMessageOptions,
    ...fieldLabelOptions
  ]
};
