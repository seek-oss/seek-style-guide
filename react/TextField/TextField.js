// @flow
import styles from './TextField.less';
import classnames from 'classnames';
import React, { Component } from 'react';
import ClearField from '../ClearField/ClearField';
import FieldMessage from '../FieldMessage/FieldMessage';
import FieldLabel from '../FieldLabel/FieldLabel';
import invoke from 'lodash/invoke';
import { TONE } from '../private/tone';
import attachRefs from '../private/attachRefs';

function combineClassNames(props = {}, ...classNames) {
  const { className, ...restProps } = props;

  return {
    className: classnames.apply(null, [...classNames, className]), // eslint-disable-line no-useless-call
    ...restProps
  };
}

type ComponentProps = {
  id: string,
  value: string,
  onChange: Function,
  onFocus?: Function,
  onBlur?: Function,
  type?: string,
  className?: string,
  valid?: boolean,
  inputProps?: Object,
  onClear?: Function
};

type FieldLabelProps = {
  label?: string,
  labelProps?: Object,
  secondaryLabel?: string,
  tertiaryLabel?: React$Node
};

type FieldMessageProps = {
  message?: false | Node | string,
  invalid?: boolean,
  help?: string,
  helpProps?: Object,
  tone?: 'positive' | 'critical' | 'neutral',
  messageProps?: Object
};

type Props = ComponentProps & FieldLabelProps & FieldMessageProps;

export default class TextField extends Component<Props> {
  static displayName = 'TextField';

  static defaultProps = { className: '', inputProps: {} };

  container: React$ElementRef<'div'>;
  input: React$ElementRef<'input'>;

  storeContainerReference = (textField: ?React$ElementRef<'div'>): void => {
    if (textField && textField !== null) {
      this.container = textField;
    }
  };

  storeInputReference = (input: React$ElementRef<'input'>): void => {
    if (input !== null) {
      this.input = input;
    }
  };

  handleMouseDownOnClear = (event: Event): void => {
    event.preventDefault(); // https://developer.mozilla.org/en/docs/Web/API/HTMLElement/focus#Notes
    invoke(this.props, 'onClear', event);
    this.input.focus();
  };

  renderInput = () => {
    const {
      id,
      value,
      onChange,
      onFocus,
      onBlur,
      type,
      inputProps = {}
    } = this.props;
    const { ref } = inputProps;
    const allInputProps = {
      id,
      value,
      onChange,
      onFocus,
      onBlur,
      type,
      ...combineClassNames(inputProps, styles.input),
      ref: attachRefs(this.storeInputReference, ref),
      'aria-describedby': `${id}-message`
    };

    return <input {...allInputProps} />;
  };

  renderClear() {
    return (
      <span
        className={styles.clearField}
        onMouseDown={this.handleMouseDownOnClear}
      >
        <ClearField />
      </span>
    );
  }

  render() {
    const {
      id,
      value,
      className,
      tone,
      onClear,
      inputProps = {},
      valid
    } = this.props;
    const resolvedValue = value || inputProps.value || '';
    const hasValue = resolvedValue.length > 0;
    const canClear = hasValue && typeof onClear === 'function';
    const classNames = classnames({
      [styles.root]: true,
      [styles.invalid]:
        typeof tone !== 'undefined' ? tone === TONE.CRITICAL : valid === false,
      [styles.canClear]: canClear,
      ...(className ? { [className]: className } : {})
    });

    const {
      label,
      labelProps,
      secondaryLabel,
      tertiaryLabel,
      invalid,
      help,
      helpProps,
      message,
      messageProps
    } = this.props;

    return (
      <div ref={this.storeContainerReference} className={classNames}>
        <FieldLabel
          {...{ id, label, labelProps, secondaryLabel, tertiaryLabel }}
        />
        {this.renderInput()}
        {this.renderClear()}
        <FieldMessage
          {...{
            id: `${id}-message`,
            invalid,
            help,
            helpProps,
            tone,
            ...(tone ? {} : { valid }),
            message,
            messageProps
          }}
        />
      </div>
    );
  }
}
