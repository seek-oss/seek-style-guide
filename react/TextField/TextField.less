@import (reference) "~seek-style-guide/theme";
@import (reference) "../ClearField/ClearFieldVariables";

.root {
  position: relative;
}

.input {
  .touchableText();
  padding: 0 @field-gutter-width;
  border: @field-border-width solid @sk-mid-gray-light;
  border-radius: @field-border-radius;
  color: @sk-charcoal;
  width: 100%;
  background-color: @sk-white;
  line-height: normal;
  -webkit-appearance: none;
  -moz-appearance: none;

  &:not([readonly]):not([disabled]):focus,
  .rootFocus & {
    outline: none;
    border-color: transparent;
    .focus();
  }

  .invalid & {
    border-color: transparent;
    box-shadow: 0 0 0 @field-border-width @sk-pink;

    &:not([readonly]):not([disabled]):focus,
    .rootFocus & {
      border-color: transparent;
      box-shadow: 0 0 0 @field-border-width @sk-pink;
    }
  }

  .canClear & {
    padding: 0 @clear-button-width 0 @field-gutter-width;
  }

  &::-webkit-search-cancel-button {
    -webkit-appearance: none;
  }
  &::-ms-clear {
    display: none;
  }
}

.clearField {
  position: absolute;
  right: 0;
  opacity: 0;
  max-width: 0;
  overflow: hidden;
  .canClear & {
    opacity: 1;
    transition: opacity 0.3s ease;
    max-width: initial;
  }
}
