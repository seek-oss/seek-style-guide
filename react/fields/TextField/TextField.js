import styles from './TextField.less';

import React, { Component, PropTypes } from 'react';
import classnames from 'classnames';

import ErrorIcon from '../../icons/ErrorIcon/ErrorIcon';
import TickCircleIcon from '../../icons/TickCircleIcon/TickCircleIcon';

import ClearField from './ClearField/ClearField';

import Text from '../../Text/Text';

function combineClassNames(props = {}, ...classNames) {
  const { className, ...restProps } = props;

  return {
    className: classnames.apply(null, [...classNames, className]), // eslint-disable-line no-useless-call
    ...restProps
  };
}

const attachRefs = (...refs) => ref => {
  refs.forEach(callRef => {
    if (typeof callRef === 'function') {
      callRef(ref);
    }
  });
};

export default class TextField extends Component {

  static displayName = 'TextField';

  static propTypes = {
    /* eslint-disable consistent-return */
    id: (props, propName, componentName) => {
      const { id, label } = props;

      if (typeof id !== 'string') {
        return new Error(`Invalid prop \`id\` of type \`${typeof id}\` supplied to \`${componentName}\`, expected \`string\`.`);
      }

      if (label && !id) {
        return new Error(`When ${componentName} has a \`label\`, it should also have an \`id\`.`);
      }
    },
    /* eslint-enable consistent-return */
    className: PropTypes.string,
    invalid: PropTypes.bool,
    label: PropTypes.string,
    /* eslint-disable consistent-return */
    labelProps: (props, propName, componentName) => {
      const { id, label, labelProps } = props;
      const { htmlFor: labelFor } = labelProps || {};

      if (typeof labelProps !== 'undefined' && typeof labelProps !== 'object') {
        return new Error(`Invalid prop \`labelProps\` of type \`${typeof labelProps}\` supplied to \`${componentName}\`, expected \`object\`.`);
      }

      if (!label && labelProps) {
        return new Error(`Specifying \`labelProps\` is redundant when \`label\` is not specified in ${componentName}.`);
      }

      if (labelFor && id) {
        return new Error(`\`labelProps.htmlFor\` will be overridden by \`id\` in ${componentName}. Please remove it.`);
      }
    },
    inputProps: (props, propName, componentName) => {
      const { id, inputProps } = props;
      const { id: inputId } = inputProps || {};

      if (typeof inputProps !== 'undefined' && typeof inputProps !== 'object') {
        return new Error(`Invalid prop \`inputProps\` of type \`${typeof inputProps}\` supplied to \`${componentName}\`, expected \`object\`.`);
      }

      if (inputId && id) {
        return new Error(`\`inputProps.id\` will be overridden by \`id\` in ${componentName}. Please remove it.`);
      }
    },
    /* eslint-disable consistent-return */
    message: PropTypes.string,
    messageProps: PropTypes.object,
    onClear: PropTypes.func
  };

  static defaultProps = {
    id: '',
    className: '',
    invalid: false,
    label: '',
    message: '',
    messageProps: {
      critical: false,
      positive: false,
      valid: false
    }
  };

  constructor() {
    super();

    this.storeInputReference = this.storeInputReference.bind(this);
    this.renderLabel = this.renderLabel.bind(this);
    this.renderInput = this.renderInput.bind(this);
    this.renderMessage = this.renderMessage.bind(this);
    this.renderMessageIcon = this.renderMessageIcon.bind(this);
    this.handleMouseDownOnClear = this.handleMouseDownOnClear.bind(this);
  }

  storeInputReference(input) {
    if (input !== null) {
      this.input = input;
    }
  }

  handleMouseDownOnClear(event) {
    const { onClear } = this.props;

    if (typeof onClear === 'function') {
      onClear(event);
      this.input.focus();
      event.preventDefault(); // https://developer.mozilla.org/en/docs/Web/API/HTMLElement/focus#Notes
    }
  }

  renderLabel() {
    const { label } = this.props;

    if (!label) {
      return;
    }

    const { labelProps, id } = this.props;
    const allLabelProps = {
      ...combineClassNames(labelProps, styles.label),
      ...(id ? { htmlFor: id } : {})
    };

    return (
      <label {...allLabelProps}>
        {label}
      </label>
    );
  }

  renderInput() {
    const { inputProps = {}, id } = this.props;
    const { ref } = inputProps;
    const allInputProps = {
      ...combineClassNames(inputProps, styles.input),
      ...(id ? { id } : {}),
      ref: attachRefs(this.storeInputReference, ref)
    };

    return (
      <input {...allInputProps} />
    );
  }

  renderClear() {
    return (
      <span
        className={styles.clearField}
        onMouseDown={this.handleMouseDownOnClear}
        onTouchStart={this.handleMouseDownOnClear}>
        <ClearField />
      </span>
    );
  }

  renderMessage() {
    const { message } = this.props;

    if (!message) {
      return;
    }

    const { critical, positive, valid, ...restMessageProps } = this.props.messageProps;
    const allMessageProps = combineClassNames(restMessageProps, styles.message);

    return (
      <Text {...allMessageProps} critical={critical} positive={positive} secondary={(!critical && !positive) || valid}>
        {this.renderMessageIcon()}
        {message}
      </Text>
    );
  }

  renderMessageIcon() {
    const { critical, positive, valid } = this.props.messageProps;

    if (critical) {
      return (
        <ErrorIcon
          filled={true}
          className={styles.messageIcon}
          svgClassName={styles.messageIconSvg}
        />
      );
    }

    if (positive || valid) {
      return (
        <TickCircleIcon
          filled={true}
          className={styles.messageIcon}
          svgClassName={styles.messageIconSvg}
        />
      );
    }

    return;
  }

  render() {
    const { className, invalid, message, onClear, inputProps = {} } = this.props;
    const hasValue = (inputProps.value && inputProps.value.length > 0);
    const canClear = hasValue && (typeof onClear === 'function');
    const classNames = classnames({
      [styles.root]: true,
      [styles.hasMessage]: message,
      [styles.invalid]: invalid,
      [styles.canClear]: canClear,
      [className]: className
    });

    return (
      <div className={classNames}>
        {this.renderLabel()}
        {this.renderInput()}
        {this.renderClear()}
        {this.renderMessage()}
      </div>
    );
  }

}
