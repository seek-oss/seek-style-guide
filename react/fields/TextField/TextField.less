@import (reference) "~seek-style-guide/theme";
@import (reference) "ClearField/ClearFieldVariables";

.root {
  display: block;
  position: relative;
}

.label {
  .standardText();
  display: inline-block;
  margin-bottom: @grid-row-height;
}

.input {
  .touchableText();
  padding: 0 @field-gutter-width;
  margin-bottom: (@grid-row-height * 3);
  border: @field-border-width solid @sk-mid-gray-light;
  border-radius: @field-border-radius;
  color: @sk-charcoal;
  width: 100%;
  background-color: @sk-white;
  -webkit-appearance: none;
  -moz-appearance: none;

  &:not([readonly]):not([disabled]):focus,
  .rootFocus & {
    outline: none;
    border-color: transparent;
    .focus();
  }

  .hasMessage & {
    margin-bottom: 0;
  }

  .invalid & {
    border-color: transparent;
    box-shadow: 0 0 0 @field-border-width @sk-pink;

    &:not([readonly]):not([disabled]):focus,
    .rootFocus & {
      border-color: transparent;
      box-shadow: 0 0 0 @field-border-width @sk-pink;
    }
  }

  .canClear & {
    padding: 0 @clear-button-width 0 @field-gutter-width;
  }

  &::-webkit-search-cancel-button {
    -webkit-appearance: none;
  }
  &::-ms-clear {
    display: none;
  }
}

.clearField {
  position: absolute;
  right: 0;
  opacity: 0;
  max-width: 0;
  overflow: hidden;
  .canClear & {
    opacity: 1;
    transition: opacity .3s ease;
    max-width: initial;
  }
}

.messageIcon {
  position: relative;
  top: 2px;
  margin-right: 4px;
}

.messageIconSvg {
  width: 14px;
  height: 14px;
  fill: currentColor;
}
