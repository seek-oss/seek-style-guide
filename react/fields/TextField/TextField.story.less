@import (reference) "~seek-style-guide/theme";

.wrapper {
  margin-bottom: (@grid-row-height * 2);
}

.label {
  .subheadingText();
  margin-bottom: @grid-row-height;
}
