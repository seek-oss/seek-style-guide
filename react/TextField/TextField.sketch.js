import React from 'react';
import TextField from './TextField';
import SketchFieldContainer from '../private/SketchFieldContainer/SketchFieldContainer';

export const symbols = {
  'TextField/Standard': (
    <SketchFieldContainer>
      <TextField id="textfield1" label="Label" message={false} />
    </SketchFieldContainer>
  ),
  'TextField/With Help Text': (
    <SketchFieldContainer>
      <TextField id="firstName2" label="Label" message="Help text" />
    </SketchFieldContainer>
  ),
  'TextField/With Critical Text': (
    <SketchFieldContainer>
      <TextField
        id="firstName3"
        label="Label"
        message="Critical text"
        tone="critical"
      />
    </SketchFieldContainer>
  ),
  'TextField/With Positive Text': (
    <SketchFieldContainer>
      <TextField
        id="firstName4"
        label="Label"
        message="Positive text"
        tone="positive"
      />
    </SketchFieldContainer>
  )
};
