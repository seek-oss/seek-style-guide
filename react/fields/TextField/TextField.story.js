import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import Story from 'Story/Story';
import styles from './TextField.story.less';

import TextField from 'fields/TextField/TextField';

storiesOf('Components', module)
  .add('TextField', () => getStory());

const getStory = () => (
  <Story title="TextField">
    <div className={styles.wrapper}>
      <h1 className={styles.label}>Default TextField</h1>
      <TextField />
    </div>
    <div className={styles.wrapper}>
      <h1 className={styles.label}>Invalid TextField</h1>
      <TextField invalid={true} message="Nup something broke" />
    </div>
  </Story>
);
