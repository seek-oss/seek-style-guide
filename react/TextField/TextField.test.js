import { shallow, mount } from 'enzyme';
import React from 'react';
import TextField from './TextField';

describe('TextField', () => {
  const requiredProps = {
    id: 'testTextField',
    onChange: () => {},
    value: ''
  };

  let spy;
  beforeEach(() => {
    spy = jest.spyOn(global.console, 'error');
  });
  afterEach(() => {
    spy.mockRestore();
  });

  it('should render with defaults', () => {
    expect(shallow(<TextField {...requiredProps} />)).toMatchSnapshot();
  });

  it('should pass through the type', () => {
    const wrapper = shallow(<TextField {...requiredProps} type="search" />);
    const inputType = wrapper.find('input').prop('type');
    expect(inputType).toEqual('search');
  });

  it('should pass through the value', () => {
    const wrapper = shallow(<TextField {...requiredProps} value="foo" />);
    const inputValue = wrapper.find('input').prop('value');
    expect(inputValue).toEqual('foo');
  });

  it('should invoke the focus handler', () => {
    const onFocus = jest.fn();
    const wrapper = shallow(<TextField {...requiredProps} onFocus={onFocus} />);
    wrapper.find('input').simulate('focus');
    expect(onFocus.mock.calls.length).toEqual(1);
  });

  it('should invoke the blur handler', () => {
    const onBlur = jest.fn();
    const wrapper = shallow(<TextField {...requiredProps} onBlur={onBlur} />);
    wrapper.find('input').simulate('blur');
    expect(onBlur.mock.calls.length).toEqual(1);
  });

  it('should invoke the change handler', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <TextField {...requiredProps} onChange={onChange} />
    );
    wrapper.find('input').simulate('change', { target: { value: 'foo' } });
    expect(onChange.mock.calls.length).toEqual(1);
    expect(onChange.mock.calls[0][0].target.value).toEqual('foo');
  });

  it('should render with input props', () => {
    expect(
      shallow(
        <TextField
          {...requiredProps}
          inputProps={{
            value: 'value',
            className: 'first-name-field',
            'data-automation': 'first-name-field'
          }}
        />
      )
    ).toMatchSnapshot();
  });

  describe('valid / tone', () => {
    it('should render the invalid style for tone="critical" even if the field is valid', () => {
      expect(
        shallow(<TextField {...requiredProps} tone="critical" valid={true} />)
      ).toMatchSnapshot();
    });

    it('should render the invalid style for valid={false} where no tone is set. It should also pass the valid prop', () => {
      expect(
        shallow(<TextField {...requiredProps} valid={false} />)
      ).toMatchSnapshot();
    });

    it('should not render the invalid style if tone is set to positive', () => {
      expect(
        shallow(<TextField {...requiredProps} valid={false} tone="positive" />)
      ).toMatchSnapshot();
    });
  });

  describe('clear button', () => {
    const handleClear = () => {};

    it('should not be visible when value is empty', () => {
      expect(
        shallow(<TextField {...requiredProps} value="" onClear={handleClear} />)
      ).toMatchSnapshot();
    });

    it('should not be visible when value is provided but no clear handler', () => {
      expect(
        shallow(<TextField {...requiredProps} value="abc" />)
      ).toMatchSnapshot();
    });

    it('should be visible when value is provided', () => {
      expect(
        shallow(
          <TextField {...requiredProps} value="abc" onClear={handleClear} />
        )
      ).toMatchSnapshot();
    });

    it('should be visible when value has white spaces only', () => {
      expect(
        shallow(
          <TextField {...requiredProps} value="  " onClear={handleClear} />
        )
      ).toMatchSnapshot();
    });

    it('should invoke the clear handler when clicked and focus on input', () => {
      const clickHandlerSpy = jest.fn();

      const wrapper = mount(
        <TextField {...requiredProps} onClear={clickHandlerSpy} />
      );
      const input = wrapper.find('input').html();
      const clearButton = wrapper.find('.TextField__clearField');

      clearButton.simulate('mouseDown');
      expect(clickHandlerSpy).toBeCalled();
      expect(global.document.activeElement.outerHTML).toEqual(input);
      clickHandlerSpy.mockRestore();
    });

    describe('inputProps', () => {
      it('should not be visible when value is empty', () => {
        expect(
          shallow(
            <TextField
              {...requiredProps}
              inputProps={{ value: '' }}
              onClear={handleClear}
            />
          )
        ).toMatchSnapshot();
      });

      it('should not be visible when value is provided but no clear handler', () => {
        expect(
          shallow(
            <TextField {...requiredProps} inputProps={{ value: 'abc' }} />
          )
        ).toMatchSnapshot();
      });

      it('should be visible when value is provided', () => {
        expect(
          shallow(
            <TextField
              {...requiredProps}
              inputProps={{ value: 'abc' }}
              onClear={handleClear}
            />
          )
        ).toMatchSnapshot();
      });

      it('should be visible when value has white spaces only', () => {
        expect(
          shallow(
            <TextField
              {...requiredProps}
              inputProps={{ value: '  ' }}
              onClear={handleClear}
            />
          )
        ).toMatchSnapshot();
      });
    });
  });
});
