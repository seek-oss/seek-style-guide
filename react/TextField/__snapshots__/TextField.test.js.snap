// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField clear button inputProps should be visible when value has white spaces only 1`] = `
<div
  className="TextField__root TextField__canClear"
>
  <FieldLabel
    id="testTextField"
    label=""
    raw={false}
    secondaryLabel=""
    tertiaryLabel=""
  />
  <input
    aria-describedby="testTextField-message"
    className="TextField__input"
    id="testTextField"
    onChange={[Function]}
    value="  "
  />
  <span
    className="TextField__clearField"
    onMouseDown={[Function]}
  >
    <ClearField />
  </span>
  <FieldMessage
    id="testTextField-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
  />
</div>
`;

exports[`TextField clear button inputProps should be visible when value is provided 1`] = `
<div
  className="TextField__root TextField__canClear"
>
  <FieldLabel
    id="testTextField"
    label=""
    raw={false}
    secondaryLabel=""
    tertiaryLabel=""
  />
  <input
    aria-describedby="testTextField-message"
    className="TextField__input"
    id="testTextField"
    onChange={[Function]}
    value="abc"
  />
  <span
    className="TextField__clearField"
    onMouseDown={[Function]}
  >
    <ClearField />
  </span>
  <FieldMessage
    id="testTextField-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
  />
</div>
`;

exports[`TextField clear button inputProps should not be visible when value is empty 1`] = `
<div
  className="TextField__root"
>
  <FieldLabel
    id="testTextField"
    label=""
    raw={false}
    secondaryLabel=""
    tertiaryLabel=""
  />
  <input
    aria-describedby="testTextField-message"
    className="TextField__input"
    id="testTextField"
    onChange={[Function]}
    value=""
  />
  <span
    className="TextField__clearField"
    onMouseDown={[Function]}
  >
    <ClearField />
  </span>
  <FieldMessage
    id="testTextField-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
  />
</div>
`;

exports[`TextField clear button inputProps should not be visible when value is provided but no clear handler 1`] = `
<div
  className="TextField__root"
>
  <FieldLabel
    id="testTextField"
    label=""
    raw={false}
    secondaryLabel=""
    tertiaryLabel=""
  />
  <input
    aria-describedby="testTextField-message"
    className="TextField__input"
    id="testTextField"
    onChange={[Function]}
    value="abc"
  />
  <span
    className="TextField__clearField"
    onMouseDown={[Function]}
  >
    <ClearField />
  </span>
  <FieldMessage
    id="testTextField-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
  />
</div>
`;

exports[`TextField clear button should be visible when value has white spaces only 1`] = `
<div
  className="TextField__root TextField__canClear"
>
  <FieldLabel
    id="testTextField"
    label=""
    raw={false}
    secondaryLabel=""
    tertiaryLabel=""
  />
  <input
    aria-describedby="testTextField-message"
    className="TextField__input"
    id="testTextField"
    onChange={[Function]}
    value="  "
  />
  <span
    className="TextField__clearField"
    onMouseDown={[Function]}
  >
    <ClearField />
  </span>
  <FieldMessage
    id="testTextField-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
  />
</div>
`;

exports[`TextField clear button should be visible when value is provided 1`] = `
<div
  className="TextField__root TextField__canClear"
>
  <FieldLabel
    id="testTextField"
    label=""
    raw={false}
    secondaryLabel=""
    tertiaryLabel=""
  />
  <input
    aria-describedby="testTextField-message"
    className="TextField__input"
    id="testTextField"
    onChange={[Function]}
    value="abc"
  />
  <span
    className="TextField__clearField"
    onMouseDown={[Function]}
  >
    <ClearField />
  </span>
  <FieldMessage
    id="testTextField-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
  />
</div>
`;

exports[`TextField clear button should not be visible when value is empty 1`] = `
<div
  className="TextField__root"
>
  <FieldLabel
    id="testTextField"
    label=""
    raw={false}
    secondaryLabel=""
    tertiaryLabel=""
  />
  <input
    aria-describedby="testTextField-message"
    className="TextField__input"
    id="testTextField"
    onChange={[Function]}
    value=""
  />
  <span
    className="TextField__clearField"
    onMouseDown={[Function]}
  >
    <ClearField />
  </span>
  <FieldMessage
    id="testTextField-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
  />
</div>
`;

exports[`TextField clear button should not be visible when value is provided but no clear handler 1`] = `
<div
  className="TextField__root"
>
  <FieldLabel
    id="testTextField"
    label=""
    raw={false}
    secondaryLabel=""
    tertiaryLabel=""
  />
  <input
    aria-describedby="testTextField-message"
    className="TextField__input"
    id="testTextField"
    onChange={[Function]}
    value="abc"
  />
  <span
    className="TextField__clearField"
    onMouseDown={[Function]}
  >
    <ClearField />
  </span>
  <FieldMessage
    id="testTextField-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
  />
</div>
`;

exports[`TextField should render with defaults 1`] = `
<div
  className="TextField__root"
>
  <FieldLabel
    id="testTextField"
    label=""
    raw={false}
    secondaryLabel=""
    tertiaryLabel=""
  />
  <input
    aria-describedby="testTextField-message"
    className="TextField__input"
    id="testTextField"
    onChange={[Function]}
    value=""
  />
  <span
    className="TextField__clearField"
    onMouseDown={[Function]}
  >
    <ClearField />
  </span>
  <FieldMessage
    id="testTextField-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
  />
</div>
`;

exports[`TextField should render with input props 1`] = `
<div
  className="TextField__root"
>
  <FieldLabel
    id="testTextField"
    label=""
    raw={false}
    secondaryLabel=""
    tertiaryLabel=""
  />
  <input
    aria-describedby="testTextField-message"
    className="TextField__input first-name-field"
    data-automation="first-name-field"
    id="testTextField"
    onChange={[Function]}
    value="value"
  />
  <span
    className="TextField__clearField"
    onMouseDown={[Function]}
  >
    <ClearField />
  </span>
  <FieldMessage
    id="testTextField-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
  />
</div>
`;

exports[`TextField valid / tone should not render the invalid style if tone is set to positive 1`] = `
<div
  className="TextField__root"
>
  <FieldLabel
    id="testTextField"
    label=""
    raw={false}
    secondaryLabel=""
    tertiaryLabel=""
  />
  <input
    aria-describedby="testTextField-message"
    className="TextField__input"
    id="testTextField"
    onChange={[Function]}
    value=""
  />
  <span
    className="TextField__clearField"
    onMouseDown={[Function]}
  >
    <ClearField />
  </span>
  <FieldMessage
    id="testTextField-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
    tone="positive"
  />
</div>
`;

exports[`TextField valid / tone should render the invalid style for tone="critical" even if the field is valid 1`] = `
<div
  className="TextField__root TextField__invalid"
>
  <FieldLabel
    id="testTextField"
    label=""
    raw={false}
    secondaryLabel=""
    tertiaryLabel=""
  />
  <input
    aria-describedby="testTextField-message"
    className="TextField__input"
    id="testTextField"
    onChange={[Function]}
    value=""
  />
  <span
    className="TextField__clearField"
    onMouseDown={[Function]}
  >
    <ClearField />
  </span>
  <FieldMessage
    id="testTextField-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
    tone="critical"
  />
</div>
`;

exports[`TextField valid / tone should render the invalid style for valid={false} where no tone is set. It should also pass the valid prop 1`] = `
<div
  className="TextField__root TextField__invalid"
>
  <FieldLabel
    id="testTextField"
    label=""
    raw={false}
    secondaryLabel=""
    tertiaryLabel=""
  />
  <input
    aria-describedby="testTextField-message"
    className="TextField__input"
    id="testTextField"
    onChange={[Function]}
    value=""
  />
  <span
    className="TextField__clearField"
    onMouseDown={[Function]}
  >
    <ClearField />
  </span>
  <FieldMessage
    id="testTextField-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
    valid={false}
  />
</div>
`;
